client/src/pages/commands/command/Form.tsx

```diff
@@ -0,0 +1,180 @@
+import TextField from '@mui/material/TextField';
+import { useFormik } from 'formik';
+import React from 'react';
+import * as yup from 'yup';
+
+import commandCreate from '~/src/api/commandCreate';
+import commandUpdate from '~/src/api/commandUpdate';
+import { CommandKeyword } from '~/src/types/CommandKeyword';
+import { CommandWithKeywords } from '~/src/types/CommandWithKeywords';
+
+type FormValues = Omit<CommandWithKeywords, 'id'>;
+
+const emptyCommand: FormValues = {
+  command: '',
+  description: '',
+  keywords: [],
+  name: '',
+};
+
+const validationSchema = yup.object().shape({
+  description: yup.string().required('Please enter a description'),
+  keywords: yup
+    .array()
+    .of(
+      yup.object().shape({
+        keyword: yup
+          .string()
+          .required('Please enter a keyword')
+          .matches(/[a-zA-Z0-9]+/, 'Only alphanumeric values allowed'),
+      })
+    )
+    .min(1, 'Please add at least one keyword'),
+  name: yup.string().required('Please give the command a name'),
+});
+
+interface CommandFormProps {
+  command?: CommandWithKeywords;
+  onError(): void;
+  onSuccess(): void;
+}
+
+const CommandForm: React.FC<CommandFormProps> = ({
+  command,
+  onError,
+  onSuccess,
+}) => {
+  const {
+    errors,
+    handleBlur,
+    handleChange,
+    handleSubmit,
+    isSubmitting,
+    setFieldValue,
+    touched,
+    values,
+  } = useFormik<FormValues>({
+    initialValues: command ?? emptyCommand,
+    onSubmit: async (submittedValues) => {
+      try {
+        if (command?.id) {
+          await commandUpdate({
+            id: command.id,
+            ...submittedValues,
+            keywords: submittedValues.keywords.map(
+              (keyword) => keyword.keyword
+            ),
+          });
+        } else {
+          await commandCreate({
+            ...submittedValues,
+            keywords: submittedValues.keywords.map(
+              (keyword) => keyword.keyword
+            ),
+          });
+        }
+        onSuccess();
+      } catch (error) {
+        onError();
+      }
+    },
+    validationSchema,
+  });
+
+  const keywordsHelperText = React.useMemo((): string | undefined => {
+    const isString = (thing: typeof errors.keywords): thing is string =>
+      typeof thing === 'string';
+    if (touched.keywords && errors.keywords) {
+      return isString(errors.keywords)
+        ? errors.keywords
+        : errors.keywords.join(', ');
+    }
+    return undefined;
+  }, [errors, touched]);
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <Card>
+        <CardContent>
+          <TextField
+            error={Boolean(errors.name)}
+            fullWidth
+            helperText={touched.name && errors.name}
+            label="Name"
+            margin="normal"
+            name="name"
+            onBlur={handleBlur}
+            onChange={handleChange}
+            value={values.name}
+          />
+          <TextField
+            fullWidth
```

Comment:
```suggestion
          <TextField
            error={Boolean(errors.description)}
            fullWidth
```

Even though this has none for now. Maybe later errors get added.